packages/lib/src/language/utils.ts

```diff
@@ -127,3 +128,21 @@ export const loadTranslations = async (locale: string, customTranslations: objec
         ...(!!customTranslations[locale] && customTranslations[locale]) // Merge with their custom locales if available
     };
 };
+
+/**
+ * Injects JSX elements in a middle of a translation and returns a JSX array
+ * The input string should use %# as the token to know where to insert the component
+ * @param translation - Translation string
+ * @param renderFunctions - An array function that renders JSX elements
+ */
+export const interpolateElement = (translation: string, renderFunctions: Array<(translation: string) => h.JSX.Element>) => {
+    // splits by regex group, it guarantees that it only splits with 2 tokens (%#)
+    const matches = translation.split(/%#(.*?)%#/gm);
+    // the map will create an array of JSX / string elements, this syntax in accepted in JSX/react to render elements
+    return matches.map((term, index) => {
+        // math to get the index of the renderFunction that should be used
+        // since we split on tokens, that means the index of the render function is half of the index of the string
+        const indexInFunctionArray = Math.floor(index / 2)
+        return index % 2 === 0 ? term : renderFunctions[indexInFunctionArray](term);
+    });
+};
```

Comment:
Just a thought: should this function test that the length of the `matches` array matches that of the `renderFunctions` array i.e that it is being passed the same number of components to render as there are place to put them?

Reply:
Follow up on this in the PR #1682